figures/scripts/fig3-panel-tp53.R

```diff
@@ -0,0 +1,182 @@
+mutators <- tmb_coding_df %>%
+  # columns of interest
+  select(Kids_First_Biospecimen_ID = Tumor_Sample_Barcode,
+         tmb) %>%
+  # keep only rows of interest
+  filter(Kids_First_Biospecimen_ID %in% tp53_scores_df_subset$Kids_First_Biospecimen_ID) %>%
+  # add a column about mutator
+  mutate(
+    mutator = case_when(
+      tmb < 10 ~ "Normal",
+      tmb >= 10 & tmb < 100 ~ "Hypermutant",
+      tmb >= 100 ~ "Ultrahypermutant")
+  )
+
+# Join data for visualization
+tp53_plot_df <- tp53_scores_df_subset %>%
+  inner_join(mutators, by="Kids_First_Biospecimen_ID") %>%
+  inner_join(
+    select(histologies_df, 
+           Kids_First_Biospecimen_ID,
+           cancer_group)
+  ) %>%
+  # filter to cancer groups and wrap for x-axis
+  filter(cancer_group %in% cancer_groups_to_plot) %>%
+  mutate(cancer_group = str_wrap(cancer_group, 20))
+
+
+#### Plot and associated legend --------------------------------------------------
+
+
+# Define colors to use
+legend_colors <- c(Normal      = "grey40",
+                   Hypermutant = "orange",
+                   Ultrahypermutant = "red")
+
+# Boxplot with overlayed jitter with colors "mapped" to `mutator`
+set.seed(9) # reproducible jitter to ensure we can see all N=6 points
+tp53_tmb_boxplot <- ggplot(tp53_plot_df) +
+  aes(
+    x = fct_reorder(cancer_group, tp53_score), # order cancer groups by tp53 score
+    y = tp53_score
+  ) +
+  geom_boxplot(
+    outlier.shape = NA, # no outliers
+    color = "grey20",    # dark grey color
+    size = 0.4 
+  ) +
+  # Separate out jitters so that the mutant layers are ON TOP OF normal
+  geom_jitter(
+    data = tp53_plot_df[tp53_plot_df$mutator == "Normal",],
+    width = 0.15, 
+    alpha = 0.7,
+    color = legend_colors["Normal"]
+  ) +
+  geom_jitter(
+    data = tp53_plot_df[tp53_plot_df$mutator == "Hypermutant",],
+    width = 0.15, 
+    pch = 21,
+    size = 2.25,
+    fill = legend_colors["Hypermutant"]
+  ) +  
+  geom_jitter(
+    data = tp53_plot_df[tp53_plot_df$mutator == "Ultrahypermutant",],
+    width = 0.15, 
+    pch = 21,
+    size = 2.25,
+    fill = legend_colors["Ultrahypermutant"]
+  ) +  
+  labs(x = "Cancer group", 
+       y = "TP53 Score") +
+  # do we want an hline at 0.5? Might be useful guiding line but also add unnecessary visual noise
+  # geom_hline(yintercept = 0.5) +
+  ggpubr::theme_pubr() +
+  theme(
+    axis.text.x = element_text(angle = 45, hjust=1, size = rel(0.8))
+  )
+
+# Export plot
+ggsave(tp53_pdf,
+       tp53_tmb_boxplot,
+       width = 8, height = 5)
+
+# Make a legend for the grey/orange/red since this was not done with normal mapping
+# For convenience, use ComplexHeatmap functions for this
+tp53_legend <- ComplexHeatmap::Legend(
+  labels = names(legend_colors),
+  legend_gp = grid::gpar(fill = legend_colors), 
+  type = "points", 
+  pch = 21
+)
+
+
+# Export legend
+pdf(tp53_legend_pdf, width = 3, height = 6)
+ComplexHeatmap::draw(tp53_legend)
+dev.off()
```

Comment:
This loosely related to my comment about the outlines of points above _because_ my knee jerk reaction to this legend was "Why are the points outlined?" and then I saw that the points are outlined in the plot itself in a way that is confusing to me. 

Anyway, the light grey background here is a bit distracting. It can be removed with Illustrator, but I think we're going to have our hands full with the figures in Illustrator already  (because it's Illustrator). So anything we can do here to help decrease that burden is a good thing!

Maybe creating a `ggplot` that does use normal mapping for the express purpose of using `get_legend()` would get it done. Side note: I don't know about the differences between `ggpubr::get_legend()` and `cowplot::get_legend()`. I believe we've used the former in this project.

Reply:
I think maybe doing all of `scale_shape_manual()`, `scale_fill_manual()`, and `scale_color_manual()` can get a normal legend in there and then separated out, as desired, with `get_legend()`.

Reply:
My idea remains to create a different plot entirely for the purpose of getting a legend. Is it ideal? No, probably not. Will it work? Hopefully!

Reply:
Yes, successful idea!